.circleci/unittest/ios/scripts/binary_ios_build.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -ex -o pipefail
+
+echo ""
+echo "DIR: $(pwd)"
+WORKSPACE=/Users/distiller/workspace
+VISION_IOS_ROOT=/Users/distiller/project/ios
+export TCLLIBPATH="/usr/local/lib"
+
+# install conda
+curl --retry 3 -o ~/conda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+chmod +x ~/conda.sh
+/bin/bash ~/conda.sh -b -p ~/anaconda
+export PATH="~/anaconda/bin:${PATH}"
+source ~/anaconda/bin/activate
+
+# install dependencies
+conda install numpy ninja pyyaml mkl mkl-include setuptools cmake cffi requests typing_extensions wget --yes
+conda install -c conda-forge valgrind --yes
+export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
+
+# sync submodules
+cd ${VISION_IOS_ROOT}
+git submodule sync
+git submodule update --init --recursive
+
+# clone main pytorch repo
+mkdir -p ${VISION_IOS_ROOT}/lib
+mkdir -p ${VISION_IOS_ROOT}/build
+git clone --recursive https://github.com/pytorch/pytorch.git
+TORCH_ROOT="${VISION_IOS_ROOT}/pytorch"
+
+# run build script
+chmod a+x ${TORCH_ROOT}/scripts/build_ios.sh
+echo "########################################################"
+cat ${TORCH_ROOT}/scripts/build_ios.sh
+echo "########################################################"
+echo "IOS_ARCH: ${IOS_ARCH}"
+echo "IOS_PLATFORM: ${IOS_PLATFORM}"
+export IOS_ARCH=${IOS_ARCH}
+export IOS_PLATFORM=${IOS_PLATFORM}
+unbuffer ${TORCH_ROOT}/scripts/build_ios.sh 2>&1 | ts
+
+LIBTORCH_HEADER_ROOT="${TORCH_ROOT}/build_ios/install/include"
```

Comment:
Do you think there would be a way of pulling the pre-compiled PyTorch binaries for iOS here, so that we avoid compiling PyTorch? Compiling PyTorch in our CI significantly increases our CI time (the ios jobs are taking 30min to run)

What we currently do for our other torchvision CIs is that we take the latest PyTorch-nightly build (that has been uploaded to conda / pip) and use it to compile torchvision. Could we do something similar here? Like taking the latest PyTorch-ios that has been uploaded to S3 and start from there?

Reply:
Yes, we can do that, but I remember you left a comment under the design doc as following:

> One thing to keep in mind when building the binaries is that the pytorch used to build torchvision should be the same as the one we make available via the pytorch binaries, otherwise there can be symbol issues.

That is the reason that I and Tao decided to build PyTorch from the source code.

Reply:
If we use the PyTorch iOS binaries from S3 to build torchvision (and the CI job times are aligned), this means that the nightlies from both torchvision and PyTorch will work, while if we pull from master this guarantee doesn't apply anymore (except if the git hashes are the same).

The comment in the doc was to try to point out exactly this: if we make available in the PyTorch nightlies a slightly different version than the one that torchvision was compiled with, there can be symbol issues.

Reply:
Fixed